.github/workflows/lifecycle-image.yml

```diff
@@ -0,0 +1,30 @@
+name: lifecycle-image
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Get metadata
+        id: get_metadata
+        run: |
+          - echo ::set-output name=LIFECYCLE_VERSION::`echo ${{ github.event.release.tag_name }} | cut -d "v" -f2`
+          - echo ::set-output name=BUILDPACK_API::`cat Makefile | grep BUILDPACK_API?= | cut -d "=" -f2`
```

Comment:
Potentially instead of reading the Makefile we could have a separate file(s) that just contains the metadata values.